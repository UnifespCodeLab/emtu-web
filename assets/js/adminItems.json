[
  { "title": "Buscas realizadas", "icon": "", "route": "/admin/buscas-realizadas" },
  { "title": "Acessos diários", "icon": "", "route": "/admin/acessos-diarios" },
  { "title": "Ranking de Linhas", "icon": "", "route": "/admin/ranking-linhas" },
  { "title": "Solicitações", "icon": "", "route": "/admin/solicitacoes" }
]