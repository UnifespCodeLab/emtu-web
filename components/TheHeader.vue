<template>
<v-card flat height="84px" tile>
    <v-toolbar height="84px" flat color="#f0f0f0">
      <v-icon id="bus-icon" @click="redirectToHomePage" aria-label="Volte para a página inicial">mdi-bus</v-icon>
      <v-toolbar-title @click="redirectToHomePage" class="ml-4 navbar-title">
        VApt
      </v-toolbar-title>
      <v-spacer />
      <v-app-bar-nav-icon id="menu-icon" @click="$emit('toggleSideBar')"  aria-label="Abrir menu de navegação"/>
    </v-toolbar>
  </v-card>
</template>

<script>
export default {
  methods: {
    redirectToHomePage () {
      this.$router.push('/')
    }
  },
  name: 'TheHeader'
}
</script>

<style scoped>
#bus-icon{
  color: #0099F0;
}
.navbar-title{
  font-family: "Lexend", "Roboto", sans-serif;
  font-size: 1.75rem;
  color: #1D1D1F;
  font-weight: bold;
  cursor: pointer;
}
#menu-icon{
  color: #1D1D1F;
}
</style>
