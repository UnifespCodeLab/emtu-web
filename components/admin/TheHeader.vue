<template>
  <v-card flat height="50px" tile>
    <v-toolbar dense>
      <v-icon color="primary">
        mdi-bus
      </v-icon>
      <v-toolbar-title class="ml-4">
        {{ routeTitle }}
      </v-toolbar-title>

      <v-spacer />
      <v-app-bar-nav-icon color="primary" @click="$emit('toggleSideBar')" />
    </v-toolbar>
  </v-card>
</template>

<script>
import adminItems from '~/assets/js/adminItems'

export default {
  name: 'TheHeader',
  computed: {
    routeTitle () {
      const currentItem = adminItems.find(item => item.route === this.$route.path)

      return currentItem.title
    }
  }
}
</script>
