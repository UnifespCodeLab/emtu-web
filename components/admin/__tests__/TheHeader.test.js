import { mount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'

import TheHeader from '~/components/admin/TheHeader.vue'
import adminItems from '~/assets/js/adminItems'

describe('Components / TheHeader', () => {
  const localVue = createLocalVue()
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()

    wrapper = mount(TheHeader, {
      localVue,
      vuetify,
      mocks: {
        $route: { path: adminItems[0].route }
      }
    })
  })

  it('should render title correctly', () => {
    const title = wrapper.findComponent({ name: 'v-toolbar-title' })
    expect(title.text()).toBe(adminItems[0].title)
    expect(wrapper.element).toMatchSnapshot()
  })
})
