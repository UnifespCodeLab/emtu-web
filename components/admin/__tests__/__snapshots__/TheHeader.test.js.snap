// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / TheHeader should render title correctly 1`] = `
<div
  class="v-card v-card--flat v-sheet theme--light rounded-0"
  style="height: 50px;"
>
  <header
    class="v-sheet theme--light v-toolbar v-toolbar--dense"
    style="height: 48px;"
  >
    <div
      class="v-toolbar__content"
      style="height: 48px;"
    >
      <i
        aria-hidden="true"
        class="v-icon notranslate mdi mdi-bus theme--light primary--text"
      />
       
      <div
        class="v-toolbar__title ml-4"
      >
        
      Home
    
      </div>
       
      <div
        class="spacer"
      />
       
      <button
        class="v-app-bar__nav-icon v-btn v-btn--icon v-btn--round theme--light v-size--default primary--text"
        color="primary"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-menu theme--light"
          />
        </span>
      </button>
    </div>
  </header>
</div>
`;
